tests/common.py

```diff
@@ -1316,12 +1316,8 @@ async def get_system_health_info(hass: HomeAssistant, domain: str) -> dict[str,
 @contextmanager
 def mock_config_flow(domain: str, config_flow: type[ConfigFlow]) -> None:
     """Mock a config flow handler."""
-    handler = config_entries.HANDLERS.get(domain)
-    config_entries.HANDLERS[domain] = config_flow
-    _LOGGER.info("Adding mock config flow: %s", domain)
-    yield
-    if handler:
-        config_entries.HANDLERS[domain] = handler
+    with patch.dict(config_entries.HANDLERS, {domain: config_flow}):
```

Comment:
We specifically added mock_config_flow to avoid using patch.dict as the latter restores the dict on exit potentially impacting clean up. There's no explanation why we think this will work now.

Reply:
https://github.com/home-assistant/core/pull/107567

As discussed on Discord. This reverts to the previous version, but always pops the mocked flow and inserts the original one (if there was any) at the end.

Reply:
Thanks!